java/test/behaviour/query/QuerySteps.java

```diff
@@ -19,308 +19,980 @@
-
-                if (matchAnswerConcept(answerIdentifier, row)) {
-                    matchingIdentifiers.add(answerIdentifier);
+    private void collectRowsAnswerIfNeeded() {
+        collectAnswerIfNeeded();
+        assertNotNull("Expected to collect ConceptRows, but the answer is not ConceptRows", collectedRows);
+    }
+
+    private void collectDocumentsAnswerIfNeeded() {
+        collectAnswerIfNeeded();
+        assertNotNull("Expected to collect ConceptDocuments, but the answer is not ConceptDocuments", collectedDocuments);
+    }
+
+    public List<Concept> getRowGetConcepts(int rowIndex) {
+        ConceptRow row = collectedRows.get(rowIndex);
+        return row.concepts().collect(Collectors.toList());
+    }
+
+    public Concept getRowGetConcept(int rowIndex, Parameters.IsByVarIndex isByVarIndex, String var) {
+        ConceptRow row = collectedRows.get(rowIndex);
+        switch (isByVarIndex) {
+            case IS:
+                return row.getIndex(row.columnNames().collect(Collectors.toList()).indexOf(var));
+            case IS_NOT:
+                return row.get(var);
+            default:
+                throw new AssertionError("Unexpected isByVarIndex: " + isByVarIndex);
+        }
+    }
+
+    public Value getRowGetValue(int rowIndex, Parameters.ConceptKind varKind, Parameters.IsByVarIndex isByVarIndex, String var) {
+        Concept concept = getRowGetConcept(rowIndex, isByVarIndex, var);
+        switch (varKind) {
+            case ATTRIBUTE:
+                return concept.asAttribute().getValue();
+            case VALUE:
+                return concept.asValue();
+            default:
+                throw new IllegalStateException("ConceptKind does not have values: " + varKind);
+        }
+    }
+
+    public boolean isConceptKind(Concept concept, Parameters.ConceptKind checkedKind) {
+        switch (checkedKind) {
+            case CONCEPT:
+                return true;
+            case TYPE:
+                return concept.isType();
+            case INSTANCE:
+                return concept.isInstance();
+            case ENTITY_TYPE:
+                return concept.isEntityType();
+            case RELATION_TYPE:
+                return concept.isRelationType();
+            case ATTRIBUTE_TYPE:
+                return concept.isAttributeType();
+            case ROLE_TYPE:
+                return concept.isRoleType();
+            case ENTITY:
+                return concept.isEntity();
+            case RELATION:
+                return concept.isRelation();
+            case ATTRIBUTE:
+                return concept.isAttribute();
+            case VALUE:
+                return concept.isValue();
+            default:
+                throw new AssertionError("Not covered ConceptKind: " + checkedKind);
+        }
+    }
+
+    // We want to call the checks on specific interfaces, not "Concept"s, so the boilerplate is needed
```

Comment:
I hate this and could write it much smaller, but it would require passing `Concept` interfaces everywhere, which is not transparent enough in terms of testing from my perspective. So, when the test says `get attribute type(p) get label`, I expect the test to literally get an instance of `AttributeType` and call its API (if it exists). 

In the end, I want to be straightforward (even if it means being really verbose) in these API checks for the drivers, otherwise, we lack coverage.